spec/arel/enhance/context_enhancer/arel_table_spec.rb

```diff
@@ -118,6 +132,16 @@
       .to eq(range_variable: false, column_reference: false, alias: true)
   end
 
+  it 'works for a table alias' do
+    sql = 'SELECT all_posts.id FROM posts AS all_posts '
+    tree = Arel.enhance(Arel.sql_to_arel(sql).first)
+
+    alias_node = '???' # HALP!
```

Comment:
@mvgijssel I've troubles figuring out how to get the alias node here

Reply:
The `TableAlias` node is created in the `visit_RangeSubselect` in the `pg_query_visitor`:

```ruby
      def visit_RangeSubselect(aliaz:, subquery:, lateral: false)
        aliaz = visit(aliaz)
        subquery = visit(subquery)
        node = Arel::Nodes::TableAlias.new(Arel::Nodes::Grouping.new(subquery), aliaz)
        lateral ? Arel::Nodes::Lateral.new(node) : node
      end
```

So the query you created should have the `TableAlias` node. To get that particular node you can do

```ruby
tree = Arel.enhance(Arel.sql_to_arel(sql).first)
tree.query(class: Arel::Nodes::TableAlias)
```

The Arel generated by Rails seems to be different than from the pg_query_visitor, so you can try something like

```
arel = Post.where(id: 1).arel.as('alias')
Arel.enhance(arel)
```

to see if the `ContextEnhancer` still breaks.